models/users.js

```diff
@@ -540,6 +545,30 @@ const getUsersBasedOnFilter = async (query) => {
   return [];
 };
 
+const getUsersWithOnboardingState = async (query) => {
+  const time = query.time;
+  const range = Number(time.split("d")[0]);
+  const { allUserStatus } = await getAllUserStatus(query);
+  const allUsersWithOnboardingState = filterUsersWithOnboardingState(allUserStatus);
+  const updatedOnboardingUsersWithDate = [];
+  const filteredUsers = [];
+  for (const user of allUsersWithOnboardingState) {
+    const result = await userModel.doc(user.userId).get();
+    filteredUsers.push(result.data());
+  }
+  filteredUsers.forEach((element) => {
+    if (element.discordJoinedAt) {
+      const userDiscordJoinedDate = new Date(element.discordJoinedAt);
+      const currentTimeStamp = new Date().getTime();
+      const timeDifferenceInMilliseconds = currentTimeStamp - userDiscordJoinedDate.getTime();
+      const currentAndUserJoinedDateDifference = Math.floor(timeDifferenceInMilliseconds / (1000 * 60 * 60 * 24));
```

Comment:
We can use Math.ceil to have an inclusive bound on upper range

Reply:
I did not understand this comment can you please explain a little bit